powerscale/provider/cluster_email_resource.go

```diff
@@ -0,0 +1,402 @@
+			"We can Create, Update and Delete the Cluster Email Settings using this resource. We can also import existing Cluster Email Settings from PowerScale array. " +
+			"Note that, Cluster Email Settings is the native functionality of PowerScale. When creating the resource, we actually load Cluster Email Settings from PowerScale to the resource state. ",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Description:         "ID of the Cluster Email Settings.",
+				MarkdownDescription: "ID of the Cluster Email Settings.",
+				Computed:            true,
+			},
+			"settings": schema.SingleNestedAttribute{
+				Description:         "Cluster email notification settings.",
+				MarkdownDescription: "Cluster email notification settings.",
+				Required:            true,
+				Attributes: map[string]schema.Attribute{
+					"batch_mode": schema.StringAttribute{
+						Description:         "This setting determines how notifications will be batched together to be sent by email.  'none' means each notification will be sent separately.  'severity' means notifications of the same severity will be sent together.  'category' means notifications of the same category will be sent together.  'all' means all notifications will be batched together and sent in a single email.",
+						MarkdownDescription: "This setting determines how notifications will be batched together to be sent by email.  'none' means each notification will be sent separately.  'severity' means notifications of the same severity will be sent together.  'category' means notifications of the same category will be sent together.  'all' means all notifications will be batched together and sent in a single email.",
+						Optional:            true,
+						Computed:            true,
+						Validators: []validator.String{
+							stringvalidator.OneOf("all", "severity", "category", "none"),
+						},
+					},
+					"mail_relay": schema.StringAttribute{
+						Description:         "The address of the SMTP server to be used for relaying the notification messages.  An SMTP server is required in order to send notifications.  If this string is empty, no emails will be sent.",
+						MarkdownDescription: "The address of the SMTP server to be used for relaying the notification messages.  An SMTP server is required in order to send notifications.  If this string is empty, no emails will be sent.",
+						Optional:            true,
+						Computed:            true,
+						Validators: []validator.String{
+							stringvalidator.RegexMatches(regexp.MustCompile(`(^$|^((([a-zA-Z0-9_][a-zA-Z0-9-]{0,61})?[a-zA-Z0-9])(\.([a-zA-Z0-9_][a-zA-Z0-9-]{0,61})?[a-zA-Z0-9])*)$|^([01]?[0-9]?[0-9]|2[0-4][0-9]|25[0-5])(\.([01]?[0-9]?[0-9]|2[0-4][0-9]|25[0-5])){3}$|^[0-9A-Fa-f]{1,4}(:[0-9A-Fa-f]{1,4}){0,5}::([0-9A-Fa-f]{1,4}(:[0-9A-Fa-f]{1,4}){0,5})?$|^[0-9A-Fa-f]{1,4}(:[0-9A-Fa-f]{1,4}){7}$)`), "Invalid SMTP server address"),
```

Comment:
for the complex regex validation, thinking better let onefs to check its value

Reply:
OK. will remove the regex check